packages/editor-ui/src/components/CredentialPicker/CredentialPicker.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+	<div>
+		<div v-if="credentialOptions.length > 0" class="dropdown">
+			<CredentialsDropdown
+				:credential-type="credentialType"
+				:credential-options="credentialOptions"
+				:selected-credential-id="selectedCredentialId"
+				@credential-selected="onCredentialSelected"
+				@new-credential="createNewCredential"
+			/>
+
+			<n8n-icon-button
+				icon="pen"
+				type="secondary"
+				:class="{
+					edit: true,
+					invisible: !selectedCredentialId,
+				}"
+				:title="$locale.baseText('nodeCredentials.updateCredential')"
+				@click="editCredential()"
+				data-test-id="credential-edit-button"
+			/>
+		</div>
+
+		<n8n-button v-else :label="`Create new ${appName} credential`" @click="createNewCredential" />
+	</div>
+</template>
+
+<script lang="ts">
```

Comment:
We should be using Composition API for all new components since we moved to Vue 3

Reply:
Converted all components to composition API